### PR TITLE
Declarative configuration missing pieces

### LINKED ISSUE #6574: Add file configuration ComponentProvider support for all SDK extension points
#6457 introduced the ComponentProvider SPI as the general purpose mechanism for providing custom SDK extension plugins for file configuration. However, the file configuration implementation now needs to be updated to actually add support for all the various SDK extension plugins. Tracking support below

- [x] SpanExporter, MetricExporter, LogRecordExporter #6493
- [x] Samplers #6494 
- [x] Propagators #6624
- [x] SpanProcessor, LogRecordProcessor #6623
- [x] Resource #6625
- [ ] Update ComponentProvider javadoc to describe usage

For resources, should likely add support for ResourceProvider instead of loading via ComponentProvider since ResourceDeteector is already a first class concept in the specification, and ResourceProvider is the java implementation of ResourceDetector.

### PR DESCRIPTION
**Pull request by jack-berg**
Resolves #6574.

Little things discovered implementing https://github.com/open-telemetry/opentelemetry-java-contrib/pull/1440.

### CODE REVIEW

File: sdk-extensions/incubator/build.gradle.kts [hunk 1]
```diff
@@ -26,6 +26,7 @@ dependencies {
 
   // io.opentelemetry.sdk.extension.incubator.fileconfig
   implementation("com.fasterxml.jackson.core:jackson-databind")
+  api("com.fasterxml.jackson.core:jackson-annotations")
```
Comment (jack-berg) [inline sdk-extensions/incubator/build.gradle.kts:29]: Without this, https://github.com/open-telemetry/opentelemetry-java-contrib/pull/1440 is full of warnings like:
```
/Users/jberg/.m2/repository/io/opentelemetry/opentelemetry-sdk-extension-incubator/1.42.0-alpha-SNAPSHOT/opentelemetry-sdk-extension-incubator-1.42.0-alpha-SNAPSHOT.jar(/io/opentelemetry/sdk/extension/incubator/fileconfig/internal/model/OpenTelemetryConfiguration.class): warning: Cannot find annotation method 'value()' in type 'JsonProperty'
```

File: sdk-extensions/incubator/src/main/java/io/opentelemetry/sdk/extension/incubator/fileconfig/FileConfiguration.java [hunk 2]
```diff
@@ -133,6 +132,17 @@ public static StructuredConfigProperties toConfigProperties(OpenTelemetryConfigu
...
+  /**
+   * Convert the {@code configuration} YAML to a generic {@link StructuredConfigProperties}.
+   *
+   * @param configuration configuration YAML
+   * @return a generic {@link StructuredConfigProperties} representation of the model
+   */
+  public static StructuredConfigProperties toConfigProperties(InputStream configuration) {
+    Object yamlObj = loadYaml(configuration, System.getenv());
+    return toConfigProperties(yamlObj);
+  }
```
Comment (jack-berg) [inline sdk-extensions/incubator/src/main/java/io/opentelemetry/sdk/extension/incubator/fileconfig/FileConfiguration.java:144]: This is an essential test utility, allowing you to write chunks of YAML, convert them to StructuredConfigProperties, and call ComponentProvider, i.e. https://github.com/open-telemetry/opentelemetry-java-contrib/pull/1440/files#diff-3d0118f231281243bf34a7e7beb3809e7a6523e8ece8a17b83f1784953389ceb

File: sdk-extensions/incubator/src/main/java/io/opentelemetry/sdk/extension/incubator/fileconfig/YamlStructuredConfigProperties.java [hunk 3]
```diff
@@ -37,14 +37,17 @@ final class YamlStructuredConfigProperties implements StructuredConfigProperties
...
   private YamlStructuredConfigProperties(
       Map<String, Object> simpleEntries,
       Map<String, List<YamlStructuredConfigProperties>> listEntries,
-      Map<String, YamlStructuredConfigProperties> mapEntries) {
+      Map<String, YamlStructuredConfigProperties> mapEntries,
+      Set<String> nullKeys) {
     this.simpleEntries = simpleEntries;
     this.listEntries = listEntries;
     this.mapEntries = mapEntries;
+    this.nullKeys = nullKeys;
```
[the remark this answers is not in the document]
Comment (jack-berg) [reply sdk-extensions/incubator/src/main/java/io/opentelemetry/sdk/extension/incubator/fileconfig/YamlStructuredConfigProperties.java:50]: Encountered in https://github.com/open-telemetry/opentelemetry-java-contrib/pull/1440/files#diff-3d0118f231281243bf34a7e7beb3809e7a6523e8ece8a17b83f1784953389ceb

Also independently included in https://github.com/open-telemetry/opentelemetry-java/pull/6549, but don't want to wait for that PR.

[the remark this answers is not in the document]
Comment (breedx-splk) [reply sdk-extensions/incubator/src/main/java/io/opentelemetry/sdk/extension/incubator/fileconfig/YamlStructuredConfigProperties.java:50]: This is a nice usability factor! Did you give any consideration to just allowing `null` values in the `simpleEntries` map? I don't think it matters too much, just noting that the ctor is a bit heavy now.

**Review by breedx-splk — APPROVED** [review]
LGTM

File: sdk-extensions/incubator/src/main/java/io/opentelemetry/sdk/extension/incubator/fileconfig/YamlStructuredConfigProperties.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (jack-berg) [reply sdk-extensions/incubator/src/main/java/io/opentelemetry/sdk/extension/incubator/fileconfig/YamlStructuredConfigProperties.java:50]: That's might be a good idea. Let me play around with it.

[the remark this answers is not in the document]
Comment (jack-berg) [reply sdk-extensions/incubator/src/main/java/io/opentelemetry/sdk/extension/incubator/fileconfig/YamlStructuredConfigProperties.java]: Yup - better.

**Pull request merged**